# Do not run MPI tests - it gives an error:
# RuntimeError: refcounted file mapping not supported on your system at /tmp/hajgato/build/PyTorch/0.3.1/intel-2018a-Python-3.6.4/pytorch-0.3.1/torch/lib/TH/THAllocator.c:525
# April 30th 2018 by B. Hajgato (Free University Brussels - VUB)
--- test/run_test.sh.orig	2018-02-09 18:07:43.000000000 +0100
+++ test/run_test.sh	2018-05-02 11:02:39.609203700 +0200
@@ -45,10 +45,11 @@
 echo "Running optim tests"
 $PYCMD test_optim.py $@
 
-echo "Running multiprocessing tests"
-$PYCMD test_multiprocessing.py $@
-MULTIPROCESSING_METHOD=spawn $PYCMD test_multiprocessing.py $@
-MULTIPROCESSING_METHOD=forkserver $PYCMD test_multiprocessing.py $@
+echo "Not Running multiprocessing tests - needs CUDA aware OpenMPI"
+#echo "Running multiprocessing tests"
+#$PYCMD test_multiprocessing.py $@
+#MULTIPROCESSING_METHOD=spawn $PYCMD test_multiprocessing.py $@
+#MULTIPROCESSING_METHOD=forkserver $PYCMD test_multiprocessing.py $@
 
 echo "Running util tests"
 $PYCMD test_utils.py $@
@@ -95,19 +96,20 @@
 BACKEND=gloo WORLD_SIZE=3 INIT_METHOD='file://'$TEMP_DIR'/shared_init_file' $PYCMD ./test_distributed.py
 distributed_tear_down
 
-if [ -x "$(command -v mpiexec)" ]; then
-  echo "Running distributed tests for the MPI backend"
-  distributed_set_up
-  BACKEND=mpi mpiexec -n 3 $PYCMD ./test_distributed.py
-  distributed_tear_down
-
-  echo "Running distributed tests for the MPI backend with file init_method"
-  distributed_set_up
-  BACKEND=mpi INIT_METHOD='file://'$TEMP_DIR'/shared_init_file' mpiexec -n 3 $PYCMD ./test_distributed.py
-  distributed_tear_down
-else
-  echo "Skipping MPI backend tests (MPI not found)"
-fi
+echo "Not Running distributed tests for the MPI backend - needs CUDA aware OpenMPI"
+#if [ -x "$(command -v mpiexec)" ]; then
+#  echo "Running distributed tests for the MPI backend"
+#  distributed_set_up
+#  BACKEND=mpi mpiexec -n 3 $PYCMD ./test_distributed.py
+#  distributed_tear_down
+#
+#  echo "Running distributed tests for the MPI backend with file init_method"
+#  distributed_set_up
+#  BACKEND=mpi INIT_METHOD='file://'$TEMP_DIR'/shared_init_file' mpiexec -n 3 $PYCMD ./test_distributed.py
+#  distributed_tear_down
+#else
+#  echo "Skipping MPI backend tests (MPI not found)"
+#fi
 
 if [[ $COVERAGE -eq 1 ]]; then
     coverage combine
